spec/services/survey/attempt_manager_spec.rb

```diff
@@ -0,0 +1,469 @@
+require 'rails_helper'
+
+describe Survey::AttemptManager do
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:survey) do
+    questions_1 = [
+      FactoryBot.build(:question, position: 1),
+      FactoryBot.build(:question, position: 2)
+    ]
+    questions_2 = [
+      FactoryBot.build(:question, position: 1),
+      FactoryBot.build(:question, position: 2)
+    ]
+    section_1 = FactoryBot.build :section, questions: questions_1, position: 1, identifier: "S1"
+    section_2 = FactoryBot.build :section, questions: questions_2, position: 2, identifier: "S2"
+    result = FactoryBot.create :survey, sections: [section_1, section_2]
+
+    (result.sections.map(&:questions).flatten).each do |question|
+      FactoryBot.create :option, question_id: question.id, position: 1, weight: 1
+      FactoryBot.create :option, question_id: question.id, position: 2, weight: 2
+    end
+
+    questions_1.first.update_attributes(skip_to_question_id: questions_1.last.id)
+    questions_1.first.options.update_all(next_question_id: questions_1.last.id)
+
+    questions_1.last.update_attributes(skip_to_question_id: questions_2.first.id)
+    questions_1.last.options.update_all(next_question_id: questions_2.first.id)
+
+    questions_2.first.update_attributes(skip_to_question_id: questions_2.last.id)
+    questions_2.first.options.update_all(next_question_id: questions_2.last.id)
+
+    questions_2.last.update_attributes(skip_to_question_id: nil)
+    questions_2.last.options.update_all(next_question_id: nil)
+
+    result
+  end
+
+  let!(:section_one) { survey.sections.where(position: 1).first }
+  let!(:section_two) { survey.sections.where(position: 2).first }
+
+  let!(:questions_section_one) { section_one.questions }
+  let!(:questions_section_two) { section_two.questions }
+
+  let(:all_questions) { questions_section_one + questions_section_two }
+  let(:current_question) { nil }
+  let(:current_section) { nil }
+
+  describe '#create_by_survey_identifier' do
```

Comment:
Great that we have this tested! Do you think its worth the extra effort to hook this up to circleci? I think a simple ruby image with bundle install && rake test should suffice

Reply:
Very good idea! Let me create a story for that

Reply:
Done: https://offcents.atlassian.net/browse/OF-1249